### ISSUE TITLE
Roles Module

### ISSUE THREAD

**Issue by dkudriavtsev** [nixnest/scepter#66]
Brainstorming area for now.

# Admin

role list

role self enable Anime # Possibly in config

role give Muted @snug#0001
role give Muted @snug#0001 1w1d
role remove Muted @snug#0001

# User

role self list

role self get Anime
role self get Anime 1w1d
role self remove Anime

# Aliases (possibly in config)

alias mute="role give Muted"
alias unmute="role remove Muted"

alias litemute="role give lite-muted"
alias unlitemute="role remove lite-muted"

mute @snug#0001
unmute @snug#0001

litemute @bitmapper#7863
unlitemute @bitmapper#7863 

# Notes

- Config may be in the config but that would might annoying
- IDs instead of names a possiblility
- Obsoletes mute module

**Comment by ldelelis** [comment]
I'd keep the "roles" module for aesthetic/low functionality roles, considering `mutes` has a certain amount of logic behind it, as to avoid having such a bloated module.

**Comment by dkudriavtsev** [comment]
some other notes:

- current mute module's logic could be adapted fairly easily for other roles
- self-roles would be basically an alias for `role give @me` with a flag to enable for specific roles

**Comment by kaewhyes** [comment]
For self-roles, including a self-ping seems unnecessary. Most bots just have you do `role <role name>` and it'll be assigned to the person that ran the command.

**Comment by dkudriavtsev** [comment]
> For self-roles, including a self-ping seems unnecessary. Most bots just have you do `role <role name>` and it'll be assigned to the person that ran the command.

In the original post, I had some examples of how self-roles would work:

```
role self list

role self get Anime
role self get Anime 1w1d
role self remove Anime
```

**Comment by kaewhyes** [comment]
> I had some examples of how self-roles would work

Yeah, this seems like the best way to assign roles. I didn't see this.
